lightning/src/ln/channelmanager.rs

```diff
@@ -4106,6 +4110,84 @@ where
 		}
 	}
 
+	/// Initiate a splice, to change the channel capacity of an existing funded channel.
+	/// TODO(splicing): Expand documentation
+	#[cfg(splicing)]
+	pub fn splice_channel(
+		&self, channel_id: &ChannelId, counterparty_node_id: &PublicKey, our_funding_contribution_satoshis: i64,
+		_our_funding_inputs: Vec<(TxIn, Transaction)>, funding_feerate_perkw: u32, locktime: u32,
+	) -> Result<(), APIError> {
+		let per_peer_state = self.per_peer_state.read().unwrap();
+
+		let peer_state_mutex = per_peer_state.get(counterparty_node_id)
+			.ok_or_else(|| APIError::ChannelUnavailable { err: format!("Can't find a peer matching the passed counterparty node_id {}", counterparty_node_id) })?;
+
+		let mut peer_state_lock = peer_state_mutex.lock().unwrap();
+		let peer_state = &mut *peer_state_lock;
+
+		// Look for the channel
+		match peer_state.channel_by_id.entry(channel_id.clone()) {
+			hash_map::Entry::Occupied(mut chan_phase_entry) => {
+				if let ChannelPhase::Funded(chan) = chan_phase_entry.get_mut() {
+					let pre_channel_value = chan.context.get_value_satoshis();
+					// Sanity check: capacity cannot decrease below 0
+					if (pre_channel_value as i64).saturating_add(our_funding_contribution_satoshis) < 0 {
+						return Err(APIError::APIMisuseError {
+							err: format!(
+								"Post-splicing channel value cannot be negative. It was {} - {}",
+								pre_channel_value, -our_funding_contribution_satoshis
```

Comment:
This line can still cause an overflow for `i64::MIN`.

Reply:
Good catch, errors within the generation of the error message are quite unpleasant.